### PR TITLE
Simplex improvements

### PR DESCRIPTION
**Pull request by TerrayTM**
Reworked simplex_iteration(), simplex(), and two_phase_simplex(), added new tests, and updated their pydocs.

### CODE REVIEW

File: pytimize/programs/_linear_program.py [hunk 1]
```diff
@@ -494,81 +501,103 @@ def __to_canonical_form(self, basis: List[int], show_steps: bool):
...
-        in_place : bool, optional (default=True)
-            Whether the operation should return a copy or be performed in place.
+        show_steps : bool (default=True)
+            Whether or not the steps should be displayed.
 
         Returns
         -------
-        result : tuple (ndarray of float, LinearProgram) # TODO update to match return format
-            The copy of the linear program or self in canonical form.
+        result : "LinearProgram"
```
Comment (jwang1000) [inline pytimize/programs/_linear_program.py:544]: Why quotes?

File: pytimize/programs/_linear_program.py [hunk 2]
```diff
@@ -718,13 +759,14 @@ def simplex_iteration(self, basis: List[int], show_steps: bool=True, in_place: b
         self.__to_canonical_form(basis, show_steps)
 
         x = self.__compute_basic_solution(basis)
-
         N = [i for i in range(self._A.shape[1]) if i not in basis]
+        k = None
```
Comment (jwang1000) [inline pytimize/programs/_linear_program.py:763]: What do these variables mean? What's going on in this section of code? Stuff like this should be commented, in addition to the method descriptions already added.

**Review by jwang1000 — APPROVED** [review]
Looks good, although more commenting apart from method docs would be good.

File: pytimize/programs/_linear_program.py (hunk 1 above)
Comment (TerrayTM) [reply pytimize/programs/_linear_program.py:544]: It's to match the function return signature. Function return typing needs quotes for classes in order for it to be valid.

**Pull request merged**